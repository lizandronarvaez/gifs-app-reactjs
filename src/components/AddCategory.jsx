/* eslint-disable react/prop-types */
import { useState } from 'react'

const AddCategory = ({ onCategoryNewValue }) => {

    const [inputValue, setInputValue] = useState("");

    const handleInputValue = ({ target }) => {
        const { value } = target;
        setInputValue(value)
    }

    const handleSubmitForm = (e) => {
        e.preventDefault();
        // Pasar el valor quitandole los espacios y pasarlo a minuscula
        const inputValueClean = inputValue.trim().toLowerCase();
        if (inputValueClean.length <= 1) return;
        // setCategories(categories => [inputValue, ...categories]);
        onCategoryNewValue(inputValueClean)
        setInputValue("");
    }
    return (
        <form onSubmit={handleSubmitForm}>
            <input
                type="text"
                name="buscador"
                placeholder='Buscar Gifs'
                value={inputValue}
                // Si solo se envie una propieda a una funcion
                // no hace falta enviar automanteicamente
                // react entiende que se pasa solo una prop
                onChange={handleInputValue}
            />
            <button type="submit">Buscar Gifs</button>
        </form>

    )
}

export default AddCategory