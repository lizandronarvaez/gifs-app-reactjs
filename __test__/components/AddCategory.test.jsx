/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react";
import AddCategory from "../../src/components/AddCategory";

describe('Pruebas en el componente <AddCategory />', () => {

    test('Realizar un snapshot al componente', () => {
        const { container } = render(<AddCategory onCategoryNewValue={() => { }} />)

        expect(container).toMatchSnapshot();
    })

    test('El valor de el input texto debe cambiar', () => {
        // Usamos el componente
        render(<AddCategory onCategoryNewValue={() => { }} />)

        // Usamos el input que queremos probar
        const input = screen.getByRole("textbox")

        // Ejecutamos el evento 
        fireEvent.input(input, { target: { value: "Lizandro" } })

        // Hacemos el test
        expect(input.value).toBe("Lizandro");
        // screen.debug()
    });

    test('Al realizar el submit deber mandar a llamarse a onCategoryNewValue', () => {
        const inputValue = "Lizandro";
        const onCategoryNewValue = jest.fn()

        render(<AddCategory onCategoryNewValue={onCategoryNewValue} />)

        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form")

        fireEvent.input(input, { target: { value: inputValue } })
        fireEvent.submit(form)

        expect(input.value).toBe("");

        // comprobar que la funcion esta siendo llamada
        expect(onCategoryNewValue).toHaveBeenCalled();
        expect(onCategoryNewValue).toHaveBeenCalledTimes(1);
    });

    test('Si el input value esta vacio,no hace la llamada a la funcion', () => {
        const onCategoryNewValue = jest.fn()

        render(<AddCategory onCategoryNewValue={onCategoryNewValue} />)

        const form = screen.getByRole("form")

        // fireEvent.input(input, { target: { value: inputValue } })
        fireEvent.submit(form)

        // expect(onCategoryNewValue).toHaveBeenCalled();
        expect(onCategoryNewValue).toHaveBeenCalledTimes(0);
        // expect(onCategoryNewValue).toHaveBeenCalled();
    });
});