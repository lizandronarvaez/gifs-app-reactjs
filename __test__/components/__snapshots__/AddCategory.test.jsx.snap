// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Realizar un snapshot al componente 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      name="buscador"
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Buscar Gifs
    </button>
  </form>
</div>
`;
